### PR TITLE
executor: migrate test-infra to testify for executor/index_lookup_join_test.go

### LINKED ISSUE #28584: migrate test-infra to testify for `index_lookup_join_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #28584

Problem Summary:

### What is changed and how it works?

executor: migrate test-infra to testify for executor/index_lookup_join_test.go

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@tisonkun @tangenta @lcwangchao PTAL

**Review by rebelice — APPROVED** [review]

File: executor/index_lookup_join_test.go [hunk 1]
```diff
@@ -394,8 +470,11 @@ PARTITIONS 1`)
 	tk.MustQuery("select /* +INL_JOIN(t1,t2) */ t1.id, t1.pc from t1 where id in ( select prefiller from t2 where t2.postfiller = 1 )").Check(testkit.Rows())
 }
 
-func (s *testSuite5) TestIssue27893(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
+func TestIssue27893(t *testing.T) {
```
Comment (tisonkun) [inline executor/index_lookup_join_test.go:473]: please move to `index_lookup_join_serial_test.go`

File: executor/index_lookup_join_test.go [hunk 2]
```diff
@@ -407,11 +486,14 @@ func (s *testSuite5) TestIssue27893(c *C) {
 	tk.MustQuery("select /*+ inl_hash_join(t2) */ count(*) from t1 join t2 on t1.a = t2.a").Check(testkit.Rows("1"))
 }
 
-func (s *testSuite5) TestPartitionTableIndexJoinAndIndexReader(c *C) {
+func TestPartitionTableIndexJoinAndIndexReader(t *testing.T) {
```
Comment (tisonkun) [inline executor/index_lookup_join_test.go:489]: please move to `index_lookup_join_serial_test.go`

**Review by tisonkun — COMMENTED** [review]
Thanks for your contribution! Comments inline.

File: executor/index_lookup_join_test.go (hunk 1 above)
Comment (hawkingrei) [reply executor/index_lookup_join_test.go:473]: add parallel

File: executor/index_lookup_join_test.go (hunk 2 above)
Comment (hawkingrei) [reply executor/index_lookup_join_test.go:490]: add parallel

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**